shell/browser/extensions/electron_extension_system.cc

```diff
@@ -84,6 +88,34 @@ void ElectronExtensionSystem::InitForRegularProfile(bool extensions_enabled) {
   app_sorting_ = std::make_unique<NullAppSorting>();
   extension_loader_ =
       std::make_unique<ElectronExtensionLoader>(browser_context_);
+
+  LoadComponentExtensions();
+}
+
+std::unique_ptr<base::DictionaryValue> ParseManifest(
+    base::StringPiece manifest_contents) {
+  JSONStringValueDeserializer deserializer(manifest_contents);
+  std::unique_ptr<base::Value> manifest = deserializer.Deserialize(NULL, NULL);
+
+  if (!manifest.get() || !manifest->is_dict()) {
+    LOG(ERROR) << "Failed to parse extension manifest.";
+    return std::unique_ptr<base::DictionaryValue>();
+  }
+  return base::DictionaryValue::From(std::move(manifest));
+}
+
+void ElectronExtensionSystem::LoadComponentExtensions() {
+  std::string utf8_error;
+  std::string pdf_manifest_string = pdf_extension_util::GetManifest();
```

Comment:
Should the variable names be made more generic in case we ever want to load other component extensions?

Reply:
As-is this is hardcoded to only load a single component extension; if we were to want to load others we should pull in more machinery from Chrome for handling them. I elided that machinery here because it seemed like overkill.